rust/processor/src/bq_analytics/mod.rs

```diff
@@ -66,3 +84,97 @@ impl From<parquet::errors::ParquetError> for ParquetProcessorError {
         ParquetProcessorError::ParquetError(err)
     }
 }
+
+pub fn create_parquet_handler_loop<ParquetType>(
+    new_gap_detector_sender: AsyncSender<ProcessingResult>,
+    processor_name: &str,
+    bucket_name: String,
+    bucket_root: String,
+    parquet_handler_response_channel_size: usize,
+    max_buffer_size: usize,
+    upload_interval: Duration,
+) -> AsyncSender<ParquetDataGeneric<ParquetType>>
+where
+    ParquetType: GetTimeStamp
+        + HasVersion
+        + HasParquetSchema
+        + NamedTable
+        + Send
+        + Sync
+        + 'static
+        + Allocative,
+    for<'a> &'a [ParquetType]: RecordWriter<ParquetType>,
+{
+    let processor_name = processor_name.to_owned();
+    let (parquet_sender, parquet_receiver) = kanal::bounded_async::<ParquetDataGeneric<ParquetType>>(
+        parquet_handler_response_channel_size,
+    );
+
+    debug!(
+        processor_name = processor_name.clone(),
+        service_type = PROCESSOR_SERVICE_TYPE,
+        "[Parquet Handler] Starting parquet handler loop",
+    );
+
+    let mut parquet_manager = GenericParquetHandler::new(
+        bucket_name.clone(),
+        bucket_root.clone(),
+        new_gap_detector_sender.clone(),
+        ParquetType::schema(),
+        upload_interval,
+        max_buffer_size,
+    )
+    .expect("Failed to create parquet manager");
+
+    tokio::spawn(async move {
+        let gcs_config = GcsClientConfig::default()
+            .with_auth()
+            .await
+            .expect("Failed to create GCS client config");
+        let gcs_client = Arc::new(GCSClient::new(gcs_config));
+
+        loop {
+            match parquet_receiver.recv().await {
+                Ok(txn_pb_res) => {
+                    let result = parquet_manager.handle(&gcs_client, txn_pb_res).await;
```

Comment:
yea this is the part you can simplify without this state... Ok/Error as state looks enough

Reply:
I initially thought knowing what has been done would give us more info when debugging. but since we have that info! when we upload I guess we can just return Ok/Error too